cli/schema_describe.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+func MakeSchemaDescribeCommand() *cobra.Command {
+	var name string
+	var root string
+	var versionID string
+
+	var cmd = &cobra.Command{
+		Use:   "describe",
+		Short: "View schema description.",
+		Long: `Introspect schema types.
+
+Example: view all schema
+  defradb client schema describe
+		
+Example: view schema by name
+  defradb client schema describe --name User
+		
+Example: view schema by root
+  defradb client schema describe --root bae123
+		
+Example: view schema by version id
+  defradb client schema describe --version bae123
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			store := mustGetStoreContext(cmd)
+
+			var schemas []client.SchemaDescription
+			switch {
+			case versionID != "":
+				schema, err := store.GetSchemaByVersionID(cmd.Context(), versionID)
+				if err != nil {
+					return err
+				}
+				schemas = []client.SchemaDescription{schema}
+
+			case root != "":
+				s, err := store.GetSchemaByRoot(cmd.Context(), root)
```

Comment:
question: Why not assign directly to `schemas` when a slice is returned?

Reply:
Compilation rules as `err` is not yet defined, the alternative would be to declare `err` outside the switch.

Reply:
`err` outside the switch would be quite fine.

Reply:
:laughing: Both are quite fine and imperfect IMO.  Declaring `err` in the broader scope has the drawback of needlessly pulling a local variable with a short lifetime into a broader scope with a longer lifetime.

Are you asking me to change this?

Reply:
No. The assignment makes no difference in the number of allocation since a slice is passed by reference. I just wanted to know what was your reasoning behind it. Now I know :)

Reply:
> The assignment makes no difference in the number of allocation since a slice is passed by reference

I wasn't talking in terms of performance, a local, short lived variable has a lower readability cost than a broadly scoped long lived one.  Especially in cases when there is no real reason for the increased scope/lifetime.

Reply:
I know :)

But if there was a significant cost to the allocation, optimizing it would win over narrowing a variable scope. Imagine thousands of schema descriptions being allocated twice. I just mentioned it here to say that it wasn't something to consider and that your reasoning was good 👍

Reply:
Ah got it :) Thanks for the explanation (and discussion) :)